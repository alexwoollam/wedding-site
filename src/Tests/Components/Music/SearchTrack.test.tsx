//ts-noscan
import '@testing-library/jest-dom';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import SearchTracks from '../../../Components/Music/SearchTrack';

Enzyme.configure({ adapter: new Adapter() });

describe('<SearchTracks/> component', () => {

    test('Component should have submission button', () => {
        render(<SearchTracks/>);
        const InfoBlockContent = screen.getByTestId('submit-button');
        expect(InfoBlockContent).toBeInTheDocument();
    });

    

    it('renders correctly', () => {
        const tree = renderer
        .create(<SearchTracks />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
