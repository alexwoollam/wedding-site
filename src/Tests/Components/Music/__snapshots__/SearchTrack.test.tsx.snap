// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchTracks/> component renders correctly 1`] = `
<div
  className="m-auto col"
>
  <button
    aria-label={null}
    className="btn btn-secondary"
    data-testid="submit-button"
    onClick={[Function]}
    type="button"
  >
    Lets rock!
  </button>
  <form
    className="m-auto max-w-md my-10"
  >
    <input
      className="p-3 block w-full rounded-md bg-gray-100 border-transparent focus:border-gray-500 focus:bg-white focus:ring-0 form-control"
      data-testid="search-input"
      id="search-track"
      onChange={[Function]}
      placeholder="Song title"
      type="text"
    />
  </form>
</div>
`;
